Updated ISY component to not overwrite state_attributes.

rmkraus opened the pull request:
**Description:**
The ISY component included an ISYDevice base class that is used by all
of the isy994 platforms. This still overwrote the state_attributes
property instead of the more appropriate device_state_attributes
property. This was also repeated in the isy994 light platform. Both of
these were addressed.

This is an alternative fix to PR #5429 

**Related issue (if applicable):** fixes #5428 

**Checklist:**
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] Tests have been added to verify that the new code works.

rmkraus committed:
Updated ISY component to not overwrite state_attributes.

The ISY component included an ISYDevice base class that is used by all
of the isy994 platforms. This still overwrote the state_attributes
property instead of the more appropriate device_state_attributes
property. This was also repeated in the isy994 light platform. Both of
these were addressed. This also fixes issue #5428.

MartinHjelmare commented on homeassistant/components/light/isy994.py:
```diff
@@ -70,7 +70,7 @@ def turn_on(self, brightness=None, **kwargs) -> None:
 
     @property
-    def state_attributes(self):
+    def device_state_attributes(self):
```
TLDR: I don't think the platform needs to report brightness in device state attributes.

The reason why this was added in https://github.com/home-assistant/home-assistant/pull/4152 was that the slider for brightness was missing for this device. But the real reason behind that behavior was actually that `state_attributes` was incorrectly overwritten in the ISYDevicemm, which your PR now fixes. The light component will report brightness attribute in state attributes when the light is on.

Also, now after https://github.com/home-assistant/home-assistant-polymer/pull/168, the brightness slider will always show in the more info panel, even when light is off, as long as the device supports brightness. So the brightness attribute is now longer connected to showing the slider or not.

rmkraus replied:
You are correct, sir. Fixed.

rmkraus committed:
Removed custom state attributes from ISY lights.

The brightness attribute need not be manually reported by the isy994
light platform.

rmkraus committed:
Removed ISY Node cleanup.

The ISY entities don’t really need to unsubscribe themselves while hass
is shutting down. Because these updates are not sent in a thread, there
is no negative impact from shutting down without unsubscribing. This
greatly speeds up hass shutdown.

rmkraus commented:
I also removed some of the ISY cleanup. It was unnecessary and really slowing down HASS as it was shutting down and cleaning up. It was also not strictly necessary.

rmkraus committed:
Removed unused attribute from isy994 light platform.

andrey-git commented:
Shouldn't light.isy994 override
```
@property
brightness()
```
?

pvizeli reviewed (dismissed):
Leave a comment

pvizeli reviewed (commented):
I think also you should remove state property from light object, that will be done from base entity and set brightness with property instead device_state_attributes

MartinHjelmare commented:
@andrey-git is correct. Good catch.

rmkraus committed:
Cleaned up ISY994 light entity class.

1) Removed the state property. This property is set in the Entity base
class and shouldn’t be overridden here.
2) Set the brightness property. This is the proper way of setting the
brightness for the Light base class.
3) Removed properties that are now unused because of these changes.

rmkraus commented:
Removed the state property and set the brightness property instead. Good catch, all.
